components/org.wso2.carbon.datasource.core/pom.xml

```diff
@@ -68,6 +68,11 @@
             <groupId>org.wso2.carbon.jndi</groupId>
             <artifactId>org.wso2.carbon.jndi</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.config</groupId>
+            <artifactId>org.wso2.carbon.config</artifactId>
+            <optional>true</optional>
```

Comment:
If we are using this in the java code (in the default flow), then we shouldn't have this dependency as optional.